src/core/searcher.rs

```diff
@@ -247,6 +247,15 @@ impl SearcherInner {
         generation: TrackedObject<SearcherGeneration>,
         doc_store_cache_size: usize,
     ) -> io::Result<SearcherInner> {
+        assert_eq!(
```

Comment:
Since the assertion does seem to protect memory safety and is quite costly (allocating a new `BTreeMap`), I think it would be better if this was a `debug_assert_eq`.

Reply:
This only happens on an Index reload.

Reply:
Fair enough, I think that makes this a nit. I would still suggest the change since the loss of robustness is minimal and it avoids adding "distributed fat" to release builds.

Reply:
I usually like to keep that kind of assert in prod. They help me sleep at night.
@shikhar your call?

Reply:
Practically: if it were a debug assertion, that may have helped as would have surfaced when trying to reproduce the issue we were seeing locally, but we also test on k8s where only release-with-debug-symbols builds are used and a debug assertion wouldn't have trigger then. So a preference for keeping this a regular assertion from me as well!